mypy/semanal.py

```diff
@@ -6849,7 +6850,9 @@ def add_symbol_table_node(
                     self.add_redefinition(names, name, symbol)
                 if not (isinstance(new, (FuncDef, Decorator)) and self.set_original_def(old, new)):
                     self.name_already_defined(name, context, existing)
-        elif name not in self.missing_names[-1] and "*" not in self.missing_names[-1]:
+        elif type_param or (
+            name not in self.missing_names[-1] and "*" not in self.missing_names[-1]
+        ):
```

Comment:
Is the issue that we aren't segregating `self.missing_names` like we do for `self.current_symbol_table`?

Reply:
If it's possible for a type parameter to get deferred couldn't it possibly mess up `class X[T = <something that causes deferral>, T]`? (or something similar w/ a bound?)

Reply:
Maybe add an `assert` for `self.add_symbol` because I think we rely on it working (in which case it returns `True`)? Better a loud crash than silently not picking up a type var IMO.